django_redis/client/default.py

```diff
@@ -59,7 +59,16 @@ def __init__(self, server, params: Dict[str, Any], backend: BaseCache) -> None:
         self._serializer = serializer_cls(options=self._options)
         self._compressor = compressor_cls(options=self._options)
 
-        self.connection_factory = pool.get_connection_factory(options=self._options)
+        # First check if local override provided, else fall back to settings
+        connection_factory_path = self._options.get(
+            "CONNECTION_FACTORY",
+            settings.get(
+                "DJANGO_REDIS_CONNECTION_FACTORY", "django_redis.pool.ConnectionFactory"
+            ),
+        )
```

Comment:
For backwards compatibility

Reply:
there is not only 'defaultClient'

Reply:
I didn't follow this comment, could you elaborate >

Reply:
Following up on this if you could clarify a bit further

Reply:
sorry for the late reply, I mean that there is not only `DefaultClient` but also others... but now I see that others all use connection factory of the parent class which is `DefaultClient`

Reply:
Also why moving this logic to `DefaultClient`? it makes testing more difficult... please move it back, add tests and changeling file